modules/ephemeris/src/main/scala/gem/horizons/HorizonsNameQuery.scala

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.horizons
+
+import gem.EphemerisKey
+
+import HorizonsNameQuery._
+
+import cats.data.EitherT
+import cats.effect.IO
+import cats.implicits._
+
+import scala.util.Either
+import scala.util.matching.Regex
+
+
+/** Horizons name-resolution query.
+  *
+  * {{{
+  * import HorizonsNameQuery.Search.Comet
+  *
+  * HorizonsNameQuery(Comet("Halley")).lookup.value.unsafeRunSync
+  * }}}
+  */
+sealed trait HorizonsNameQuery[A] {
+
+  /** URL string corresponding the the horizons request. */
+  def urlString: String
+
+  /** Returns a program that will perform the name lookup when executed,
+    * returing a List of ephemeris keys with their common name.
+    */
+  def lookup: Result[List[Resolution[A]]]
+}
+
+object HorizonsNameQuery {
+
+  /** Describes a non-sidereal target query of a specific type of object. */
+  sealed abstract class Search[A](val queryString: String) extends Product with Serializable
+
+  object Search {
+    final case class Comet(partial: String)     extends Search[EphemerisKey.Comet    ](s"NAME=$partial*;CAP")
+    final case class Asteroid(partial: String)  extends Search[EphemerisKey.Asteroid ](s"ASTNAM=$partial*"  )
+    final case class MajorBody(partial: String) extends Search[EphemerisKey.MajorBody](s"$partial"          )
+  }
+
+  sealed trait Error extends Product with Serializable
+
+  final case class HorizonsException(e: Throwable)                  extends Error
```

Comment:
Not used, yet.  I think it would only be needed if the client wants to map an error running the service.